src/ray/protobuf/autoscaler.proto

```diff
@@ -0,0 +1,172 @@
+/*
+This file defines an experimental interface designed for Ray
+autoscalers.
+
+It is currently unstable and under active development, but it is
+intended to become stable and forwards-compatible in the future.
+*/
+
+
+syntax = "proto3";
+option cc_enable_arenas = true;
+package ray.rpc;
+
+message ResourceBundle {
+  // The fixed point resource size of the resource (granularity: 1/10000).
+  map<string, int64> resources = 1;
+}
+
+// Type of placement group strategy.
+enum PlacementStrategy {
+  // Packs Bundles into as few nodes as possible.
+  PACK = 0;
+
+  // Places Bundles across distinct nodes or processes as even as possible.
+  SPREAD = 1;
+
+  // Packs Bundles within one node. The group is not allowed to span multiple nodes.
+  STRICT_PACK = 2;
+
+  // Places Bundles across distinct nodes.
+  // The group is not allowed to deploy more than one bundle on a node.
+  STRICT_SPREAD = 3;
+}
+
+enum SourceType {
+  // A task that (assuming no deadlock can eventually be scheduled on the cluster.
+  TASK = 0;
+
+  // A actor that (assuming no deadlock can eventually be scheduled on the cluster.
+  ACTOR = 1;
+
+  // A placement group that (assuming no deadlock can eventually be scheduled on the cluster.
+  PLACEMENT_GROUP = 2;
+
+  // Represents a job's call to
+  // `ray.autoscaler.sdk.request_resources()`. This type requests
+  // resources to be available on the cluster but won't actually
+  // utilize them.
+  SDK_RESOURCE_REQUEST = 3;
+}
+
+// A workload (task/actor/placement group) unfulfilled by the the cluster's internal scheduler(s).
+message QueuedResources {
+  // The number of resources of this quantity. This should be
+  // functionally equivalent to setting quantity=1 and repeating the
+  // field many times.
+  uint32 quantity = 1;
+
+  // Whether the queued resources can eventually be scheduled on the cluster
+  // (neglecting deadlock).
+  bool is_feasible = 2;
+
+  // The source of this queued resource.
+  SourceType type = 3;
+
+  // If the queued resource represents an actor or task, there is
+  // 1 entry in `bundles`. In a placement group, each bundle
+  // receives its own entry.
+  repeated ResourceBundle bundles = 4;
+
+  // The placement strategy of bundles within this queued resources.
+  PlacementStrategy placement_strategy = 5;
```

Comment:
do we care about pending time?

Reply:
the autoscaler currently doesn't have/use that information, but if you think it's good for future proofing, we can add it.

Reply:
Ooh, yeah, how long a demand has been pending would be a great thing to take into account when autoscaling.
IMO, better left for later.